### ISSUE TITLE
Incorrect execution result for simple program

### ISSUE THREAD

**Issue by jserv** [jserv/amacc#60]
[maze.c.gz](https://github.com/jserv/amacc/files/4908831/maze.c.gz) is a simple maze generator written by Joe Wingbermuehle. The modified program can be parsed and executed by AMaCC. However, the program output is NOT correct. It was assumed to be kind of maze though. Also, the functionality to solve maze was incorrect.

**Comment by jserv** [comment]
After applying the potential fix #61 against modulo operator and replacing all occurrences of `%` with `mod` call, program `maze.c` is known to work.

```
$ /usr/bin/qemu-arm -L /usr/arm-linux-gnueabihf ./amacc  maze.c
[]  [][][][][][][][][][][][][][][][][][][][][]
[]      []              []                  []
[][][]  [][][]  []  [][][]  [][][][][]  []  []
[]          []  []          []      []  []  []
[]  [][][]  []  [][][][][]  []  []  [][][]  []
[]  []      []          []  []  []      []  []
[]  []  [][][][][][][]  [][][]  []  []  []  []
[]  []                  []      []  []      []
[]  [][][][][][][][][][][]  [][][][][]  [][][]
[]  []                  []  []      []      []
[]  []  [][][][][][][][][]  [][][]  [][][]  []
[]  []                  []      []  []  []  []
[]  [][][][][][][][][]  []  []  []  []  []  []
[]                  []  []  []      []      []
[][][][][][][][][]  []  []  [][][][][][][][][]
[]              []  []  []                  []
[]  [][][][][]  []  []  []  [][][][][][][]  []
[]  []      []  []      []              []  []
[]  [][][]  []  []  [][][][][][][][][]  []  []
[]      []      []      []          []  []  []
[]  []  [][][][][][][][][]  []  []  [][][]  []
[]  []                      []  []          []
[][][][][][][][][][][][][][][][][][][][][]  []
```

**Comment by jserv** [comment]
Close via commit 770e4c966dec76d571187d8c310d8ee625aa66bf
